### PR TITLE
Use charstring as plain text when lengthIV is -1.

### LINKED ISSUE #7769: Strange box characters in PDF
I see a problem with a particular PDF when viewed on the latest version of pdf.js, e.g. through your viewer here:

https://mozilla.github.io/pdf.js/web/viewer.html

This is the PDF displayed using Preview on a Mac:

screen shot 2016-10-31 at 14 18 01

But this is what I see in all browsers I've tried:

screen shot 2016-10-31 at 14 15 36

Unfortunately I'm not sure what software is used to create the PDF.

The PDF is here:

[Parish-Bulletin-10-22.1 (1).pdf](https://github.com/mozilla/pdf.js/files/561923/Parish-Bulletin-10-22.1.1.pdf)

I hope this helps. Thank you!

Dan

### LINKED ISSUE #8644: can't render this pdf file correctly
Link to PDF file (or attach file here):
[bad.pdf](https://github.com/mozilla/pdf.js/files/1145277/bad.pdf)

Configuration:

- Web browser and its version: chrome 58.0.0
- Operating system and its version:win7
- PDF.js version:1.0.0
- Is an extension:no

Steps to reproduce the problem:
1. copy this file to pdfjs directory
2.view it in the browser  , location like  'http://localhost:8888/web/viewer.html?file=bad.pdf'

What is the expected behavior? (add screenshot)
image

What went wrong? (add [screenshot)]
image

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

### PR DESCRIPTION
**Pull request by brendandahl**
Fixes #7769.
Fixes #8644.

### CODE REVIEW

File: src/core/type1_parser.js [hunk 1]
```diff
@@ -150,6 +150,10 @@ var Type1CharString = (function Type1CharStringClosure() {
                 break;
               }
               subrNumber = this.stack.pop();
+              if (!subrs[subrNumber]) {
```
Comment (brendandahl) [inline src/core/type1_parser.js]: This isn't necessary to fix the broken PDF, but it something I noticed when we were previously trying to decrypt the charstring.

**Comment by brendandahl** [comment]
Also fixes #8644

**Review by Snuffleupagus — APPROVED** [review]
Really nice find, thanks for fixing this!

**Pull request merged**